tests/test_freeimage.py

```diff
@@ -3,35 +3,65 @@
 
 import os
 import sys
+import shutil
 
 import numpy as np
 
 from pytest import raises, skip
 import pytest
-from imageio.testing import run_tests_if_main, get_test_dir, need_internet
 
 import imageio
 from imageio import core
-from imageio.core import get_remote_file, IS_PYPY
+from imageio.core import IS_PYPY
 
 
-test_dir = get_test_dir()
+@pytest.fixture(scope="module")
+def setup_library(tmp_path_factory, image_cache):
 
+    # Checks if freeimage is installed by the system
+    from imageio.plugins.freeimage import fi
+
+    use_imageio_binary = False
+    if not fi.has_lib():
+        use_imageio_binary = True
 
-def setup_module():
     # During this test, pretend that FI is the default format
     imageio.formats.sort("-FI")
 
-    # This tests requires our version of the FI lib
-    try:
-        imageio.plugins.freeimage.download()
-    except core.InternetNotAllowedError:
-        # We cannot download; skip all freeimage tests
-        need_internet()
-
-
-def teardown_module():
-    # Set back to normal
+    if use_imageio_binary:
+
+        # Setup from image_cache/freeimage
+        ud = tmp_path_factory.getbasetemp() / "userdir"
+        if sys.platform.startswith("win"):
+            if "LOCALAPPDATA" in os.environ:  # saves it
+                os.environ["OLD_LOCALAPPDATA"] = os.environ["LOCALAPPDATA"]
+            os.environ["LOCALAPPDATA"] = str(ud)
+        else:
+            os.environ["IMAGEIO_USERDIR"] = str(ud)
+        os.makedirs(ud, exist_ok=True)
+        add = core.appdata_dir("imageio")
+        os.makedirs(add, exist_ok=True)
+        shutil.copytree(image_cache / "freeimage", os.path.join(add, "freeimage"))
+        fi.load_freeimage()
+        assert fi.has_lib(), "imageio-binaries' version of libfreeimage was not found"
+
+    yield
+
+    if use_imageio_binary:
+
+        if sys.platform.startswith("win"):
+            del os.environ["LOCALAPPDATA"]
+            if "OLD_LOCALAPPDATA" in os.environ:
+                os.environ["LOCALAPPDATA"] = os.environ["OLD_LOCALAPPDATA"]
+                del os.environ["OLD_LOCALAPPDATA"]
+        else:
+            del os.environ["IMAGEIO_USERDIR"]
+
+        # on windows, loaded DLLs may refuse to get deleted
+        # there is no clean way to release those, so we leave it there
+        shutil.rmtree(ud, ignore_errors=True)
```

Comment:
```suggestion
```

DLLs won't get deleted, because they are still loaded. You can force them to unload, but I don't know how to do that in python (never had to). Either way, we can let pytest take care of cleaning up :)

Reply:
Very well - I'd prefer that as well.

Reply:
I'm removing all code that cleans up.